src/legacy/core_plugins/vega/public/vega_request_handler.js

```diff
@@ -17,28 +17,26 @@
  * under the License.
  */
 
+import chrome from 'ui/chrome';
+import { timefilter } from 'ui/timefilter';
+import { buildEsQuery, getEsQueryConfig } from '@kbn/es-query';
+
 import { VegaParser } from './data_model/vega_parser';
 import { SearchCache } from './data_model/search_cache';
 import { TimeCache } from './data_model/time_cache';
-import { timefilter } from 'ui/timefilter';
-import { buildEsQuery, getEsQueryConfig } from '@kbn/es-query';
 
-export function VegaRequestHandlerProvider(es, serviceSettings, config, esShardTimeout) {
+export function createVegaRequestHandler(es, serviceSettings) {
+  const uiSettings = chrome.getUiSettingsClient();
```

Comment:
`uiSettings` I believe is already available on `core` object, that `setup()` lifecycle of plugin should receive as the first argument.